client/lib/preferences/actions.js

```diff
@@ -36,7 +37,12 @@ function mergePreferencesToLocalStorage( preferences ) {
 }
 
 PreferencesActions.fetch = function() {
-	var localStorage = getLocalStorage();
+	var localStorage = getLocalStorage(),
+		userIsLoggedIn = Boolean( user.data );
```

Comment:
So yes, I opted to check `user.data` instead of `user.get()` because this function actually [fetches the user profile](https://github.com/Automattic/wp-calypso/blob/2a0b41d694a589947c43da4861a1e254618ebbc7/client/lib/user/user.js#L105) if this user is not logged in, hence resulting in an additional and unnecessary request. Maybe we should update the `Signup` component to follow the same approach?